test/sshkit/scp_functional_test.exs

```diff
@@ -1,3 +1,178 @@
 defmodule SSHKit.SCPFunctionalTest do
   use SSHKit.FunctionalCase, async: true
+
+  alias SSHKit.SCP
+  alias SSHKit.SSH
+
+  @defaults [silently_accept_hosts: true]
+
+  describe "upload/4" do
+    @tag boot: 1
+    test "sends a file", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace/local_file.txt"
+      remote = "file.txt"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote)
+      assert verify_transfer(conn, local, remote)
+      :ok = SSH.close(conn)
+    end
+
+    @tag boot: 1
+    test "recursive: true", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace"
+      remote = "/home/#{host.user}/destination"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote, recursive: true)
+      assert verify_transfer(conn, local, remote)
+      :ok = SSH.close(conn)
+    end
+
+    @tag boot: 1
+    test "preserve: true", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace/local_file.txt"
+      remote = "file.txt"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote, preserve: true)
+      assert verify_mode(conn, local, remote)
+      # assert verify_atime(conn, local, remote)
```

Comment:
Just took a look at this again, and yeah, the issue still exists where atime doesn't ever match. I will just remove the `atime` tests for now in the interest of moving this PR forward